File: addons/guide-cpp/src/triggers/guide_trigger_down.cpp
#include "guide_trigger_down.h"

using namespace godot;

void GUIDETriggerDown::_bind_methods() {
}

GUIDETriggerDown::GUIDETriggerDown() {
}

GUIDETriggerDown::~GUIDETriggerDown() {
}

bool GUIDETriggerDown::is_same_as(const Ref<GUIDETrigger> &other) const {
    Ref<GUIDETriggerDown> o = other;
    if(o.is_null()) return false;
    return true;
}

GUIDETrigger::GUIDETriggerState GUIDETriggerDown::_update_state(Vector3 input, double delta, GUIDEAction::GUIDEActionValueType value_type) {
    if (_is_actuated(input, value_type)) {
        return TRIGGERED;
    }
    return NONE;
}

String GUIDETriggerDown::_editor_name() const {
    return "Down";
}

String GUIDETriggerDown::_editor_description() const {
    return "Fires, when the input exceeds the actuation threshold. This is\nthe default trigger when no trigger is specified.";
}

 // namespace godot

File: addons/GuideCS/Triggers/GUIDETriggerDown.cs
using System;
using Godot;

namespace Guide;

[Tool]
[GlobalClass]
public partial class GUIDETriggerDown : GUIDETrigger {
	public override bool IsSameAs(GUIDETrigger other) {
		return other is GUIDETriggerDown;
	}

	public override GUIDETriggerState _UpdateState(Vector3 input, float delta, GUIDEAction.GUIDEActionValueType valueType) {
		if (_IsActuated(input, valueType)) {
			return GUIDETriggerState.Triggered;
		}
		return GUIDETriggerState.None;
	}

	public override string _EditorName() => "Down";

	public override string _EditorDescription() => "Fires, when the input exceeds the actuation threshold. This is\nthe default trigger when no trigger is specified.";
}
